packages/fxa-settings/src/pages/Signup/Confirm/index.tsx

```diff
@@ -32,25 +65,128 @@ const Confirm = ({ email }: RouteComponentProps & ConfirmProps) => {
     instructionText: `Check your email for the confirmation link sent to ${email}`,
   };
 
-  const resendEmailHandler = async () => {
+  // Verifies if we have access to profile info and session token.
+  // If we do, set the email and render the page. If not, we don't know
+  // who the user is and can't confirm if we've sent a confirmation
+  // link, and we can't send a new one without an email address => redirect
+  // the user to retry signup.
+  useEffect(() => {
+    try {
+      if (sessionTokenId) {
+        getEmail();
+      } else {
+        redirectToSignup();
+      }
+    } catch (e) {
+      redirectToSignup();
+    }
+  }, [getEmail, sessionTokenId]);
+
+  useEffect(() => {
+    if (email) {
+      setShowLoadingSpinner(false);
+    }
+  }, [email, setShowLoadingSpinner]);
+
+  const navigateToConfirmCode = useCallback(
+    () =>
+      navigate('/confirm_signup_code?showReactApp=true', {
```

Comment:
Q: if we roll out to n% population, would this force 100% of the people into the React page cohort? Or should we omit the query param?

Possibly not relevant since I think we have `?showReactApp=` redirects a couple other places in the codebase:
```sh
git grep -n "?showReactApp=" packages | grep -Ev "\.(spec|test)\." | grep -v "/tests/"

packages/fxa-settings/src/pages/ResetPassword/ConfirmResetPassword/index.tsx:48:    navigate('reset_password?showReactApp=true', { replace: true });
packages/fxa-settings/src/pages/ResetPassword/index.tsx:82:      navigate('confirm_reset_password?showReactApp=true', {
packages/fxa-settings/src/pages/Signup/Confirm/index.tsx:93:      navigate('/confirm_signup_code?showReactApp=true', {
```

Reply:
The flag here is mostly for development, to make sure that we stay within the react app if we refresh the page. It shouldn't affect users in experimentation groups once rolled out (though we should still remove these flags once we're ready to go live), since they wouldn't land on the initial React /confirm page in the first place unless they're already enrolled in the React group.